openedx/core/djangoapps/content/search/tests/test_api.py

```diff
@@ -407,6 +408,56 @@ def test_index_library_block_tags(self, mock_meilisearch):
             any_order=True,
         )
 
+    @override_settings(MEILISEARCH_ENABLED=True)
+    def test_index_library_block_collections(self, mock_meilisearch):
+        """
+        Test indexing an Library Block that is in two collections.
+        """
+        collection1 = authoring_api.create_collection(
+            learning_package_id=self.library.learning_package.id,
+            key="COL1",
+            title="Collection 1",
+            created_by=None,
+            description="First Collection",
+        )
+
+        collection2 = authoring_api.create_collection(
+            learning_package_id=self.library.learning_package.id,
+            key="COL2",
+            title="Collection 2",
+            created_by=None,
+            description="Second Collection",
+        )
+
+        # Add Problem1 to both Collections (these internally call `upsert_block_collections_index_docs`)
+        # (adding in reverse order to test sorting of collection tag))
+        for collection in (collection2, collection1):
+            library_api.update_library_collection_components(
+                self.library.key,
+                collection_key=collection.key,
+                usage_keys=[
+                    self.problem1.usage_key,
+                ],
+            )
+
+        # Build expected docs with collections at each stage
+        doc_problem_with_collection2 = {
+            "id": self.doc_problem1["id"],
+            "collections": [collection2.key],
+        }
+        doc_problem_with_collection1 = {
+            "id": self.doc_problem1["id"],
+            "collections": [collection1.key, collection2.key],
+        }
+
+        mock_meilisearch.return_value.index.return_value.update_documents.assert_has_calls(
+            [
+                call([doc_problem_with_collection2]),
+                call([doc_problem_with_collection1]),
+            ],
+            any_order=True,
```

Comment:
I think we can have a more deterministic approach here to assert we don't have extra calls
```suggestion
        self.assertEqual(
            mock_meilisearch.return_value.index.return_value.update_documents.call_count,
            2,
        )
        mock_meilisearch.return_value.index.return_value.update_documents.assert_has_calls(
            [
                call([doc_problem_with_collection2]),
                call([doc_problem_with_collection1]),
            ],
```

Reply:
Sure -- we still can't rely on the order these calls are made in, but I've added asserts for the `call_count`, e.g. https://github.com/openedx/edx-platform/pull/35469/commits/a81ea9afd20a27918f376983f65cb7ee1c32667c#diff-f6238e12f564d80e4a90b6f385c31c1396de25784be34992c1f17d1c9ca557a1R353